src/mig/modules/pkgprint/templates.go

```diff
@@ -0,0 +1,91 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Contributor: Aaron Meihm ameihm@mozilla.com [:alm]
+package pkgprint
+
+import (
+	"fmt"
+	"regexp"
+)
+
+type ppTemplate struct {
+	name        string
+	fingerprint fingerprint
+}
+
+type fingerprint struct {
+	filename     string
+	isRegexp     bool
+	pathFilter   string
+	contentMatch string
+	transform    func(string) (string, error)
+	forceRoot    string
+}
+
+var templates = []ppTemplate{
+	{"mediawiki", fingerprint{
+		"DefaultSettings.php",
+		false,
+		"",
+		".*wgVersion = (\\S+)",
+		transformNull,
+		"",
+	}},
+	{"django", fingerprint{
+		"__init__.py",
+		false,
+		"django",
+		"^VERSION = (\\S+, \\S+, \\S+, \\S+, \\S+)",
+		transformDjango,
+		"",
+	}},
+	{"linuxkernel", fingerprint{
+		"version",
+		false,
+		"",
+		"^(Linux version.*)",
+		transformNull,
+		"/proc",
+	}},
+	{"linuxmodules", fingerprint{
+		"modules",
+		false,
+		"",
+		"^(\\S+)",
+		transformNull,
+		"/proc",
+	}},
+	{"pythonegg", fingerprint{
+		"PKG-INFO",
+		false,
+		"egg-info",
+		"^Version: (\\S+)",
+		transformNull,
+		"",
+	}},
+}
+
+func getTemplateFingerprint(name string) *fingerprint {
```

Comment:
Why not make `templates` a `map[string]fingerprint` where `string` is the ppTemplate.name field? That way, you could locate a template using a key lookup instead of iterating.